contracts/move_funds/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+extern crate bincode;
+extern crate solana;
+
+use bincode::deserialize;
+use solana::system_contract::AccountInfo;
+
+#[no_mangle]
+pub extern "C" fn process(infos: &mut Vec<AccountInfo>, data: &[u8]) {
+    let tokens: i64 = deserialize(data).unwrap();
+    if infos[0].account.tokens >= tokens {
+        infos[0].account.tokens -= tokens;
+        infos[1].account.tokens += tokens;
+    } else {
+        println!(
+            "Insufficient funds, asked {}, only had {}",
+            tokens, infos[0].account.tokens
+        );
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bincode::serialize;
+    use solana::bank::Account;
+    use solana::signature::Pubkey;
+
+    #[test]
+    fn test_move_funds() {
+        let tokens: i64 = 100;
+        let data: Vec<u8> = serialize(&tokens).unwrap();
+        let keys = vec![Pubkey::default(); 2];
+        let mut accounts = vec![Account::default(), Account::default()];
+        accounts[0].tokens = 100;
+        accounts[1].tokens = 1;
```

Comment:
What happens if this is zero?

Reply:
'this' being the value we assign to `accounts[1].tokens'?

Reply:
yes

Reply:
then the result of that test would be that accounts[0].tokens == 0 and accounts[1].tokens == 100.  What issue did you notice?